### PR TITLE
Feature/matrix tile

### PR DESCRIPTION
**Pull request by maddyscientist**
This pull request is an initial, and yet significant, rework of the multigrid setup kernels to better express locality in the form of explicit register-level tiling. 
* Introduced `MatrixTile` infrastructure for optimized deployment of matrix-matrix type computations, as performed in the multigrid setup kernels
* `computeUV`, `computeVUV` and `computeYhat` kernels all rewritten to use this new framework
* Verified against both staggered and Wilson multigrid setups
* As part of this pull, I have updated to the latest official jitify snapshot (and also verified the multigrid setup runs through fine with a jitify-enabled build)
* Improved jitify verbosity setting: when doing a `DEVEL` build of QUDA we can now see the build logs at runtime
* Reduced tuning time for plaquette (narrow grid search space which was unnecessarily large previously)

The end result of this work is that the multigrid-setup kernels are now significantly faster, while also autotuning much faster.  For some pathological multigrid setups, we see an 8x reduction in tuning time and 10x boost in performance.

For an example of this performance boost, we consider the following invocation
```bash
multigrid_invert_test --dim 16 16 16 16 --mg-levels 3 --mg-nvec 0 24 --mg-nvec 1 32 --prec-precondition half --prec-null half --nsrc 0
```
This will run 3-level MG and only run the setup.  Kernel performance is as follows (perf in GFLOPS):

  | old | new
-- | -- | --
UV level 0 | 1630 | 1660
VUV level 0 | 1890 | 2950
Yhat level 0 | 3110 | 5900
Yhat level 0 max | 3170 | 6730
UV level 1 | 340 | 4100
VUV level 1 | 3000 | 3400
Yhat level 1 | 830 | 2660
Yhat level 1 max | 810 | 4830

Noting that the `MatrixTile` framework is far from complete and is a stepping stone towards tensor-core deployment more broadly in QUDA.

### CODE REVIEW

**Comment by maddyscientist** [comment]
Additional note: I have done some rudimentary tuning of the tile-size deployed in each of the kernels to pick a near-optimal tiling depending on the number of degrees of freedom.  The tile sizes are all statically checked to ensure they are valid (the tile size must be a divisor of the matrix size).

For example, here is the tile size setting used for `computeUV`
```c++
// tile used for computeUV
static constexpr int tile_height_uv = fineColor % 4 == 0 ? 4 : fineColor % 3 == 0 ? 3 : fineColor % 2 ? 2 : 1;
static constexpr int tile_width_uv = coarseColor % 2 == 0 ? 2 : 1;
TileSize<fineColor, coarseColor, fineColor, tile_height_uv, tile_width_uv, 1> uvTile;
```

**Comment by weinbe2** [comment]
Beginning to review this PR. Should get done over the course of the day.

**Comment by weinbe2** [comment]
This is largely good. Do not merge yet, @maddyscientist identified a segfault that she's addressing as her time permits today/tomorrow. Once that's through this should be good to go.

**Review by weinbe2 — APPROVED** [review]
All's looking great, I'm very excited about this PR.

**Comment by weinbe2** [comment]
I'll merge pending the Jenkins build tests completing.

**Pull request merged**
